src/core/core.layouts.js

```diff
@@ -48,10 +94,141 @@ function findMaxPadding(boxes) {
 	};
 }
 
-function addSizeByPosition(boxes, size) {
-	helpers.each(boxes, function(box) {
-		size[box.position] += box.isHorizontal() ? box.height : box.width;
-	});
+function addSizeByPosition(outerDims, boxes) {
+	var result = extend({}, outerDims);
+	var i, ilen, layout, box;
+	for (i = 0, ilen = boxes.length; i < ilen; ++i) {
+		layout = boxes[i];
+		box = layout.box;
+		result[layout.pos] += layout.horizontal ? box.height : box.width;
+	}
+	return result;
+}
+
+function calculateMinSizes(boxes, params) {
+	var chartArea = params.chartArea;
+	var i, ilen, layout, box, minSize;
+
+	for (i = 0, ilen = boxes.length; i < ilen; ++i) {
+		layout = boxes[i];
+		box = layout.box;
+
+		if ((layout.horizontal = box.isHorizontal())) {
+			minSize = box.update(box.fullWidth ? params.fullWidth : chartArea.w, params.hBoxMaxHeight);
```

Comment:
I think `params.fullWidth` and `chartArea.w` have to be the same here since `updateWithMaxPadding` hasn't been called yet. Can we replace `box.fullWidth ? params.fullWidth : chartArea.w` with `chartArea.w`